src/main/java/blackjack/domain/WinningDiscriminator.java

```diff
@@ -0,0 +1,62 @@
+package blackjack.domain;
+
+import blackjack.domain.gambler.Name;
+
+import java.util.Arrays;
+import java.util.Map;
+
+import static blackjack.domain.WinningType.*;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+
+public class WinningDiscriminator {
+    public static final int BLACK_JACK = 21;
+    private final int dealerScore;
+    private final Map<Name, Integer> playerScores;
+
+    public WinningDiscriminator(final int dealerScore, final Map<Name, Integer> playerScores) {
+        this.dealerScore = dealerScore;
+        this.playerScores = playerScores;
+    }
+
+    public Map<Name, WinningType> judgePlayersResult() {
+        return playerScores.keySet()
+                .stream()
+                .collect(toMap(identity(), this::judgePlayerResult));
+    }
+
+    public Map<WinningType, Integer> judgeDealerResult() {
+        Map<WinningType, Integer> winningResult = createWinningResult();
+        for (final Name playerName : playerScores.keySet()) {
+            countDealerWinning(playerName, winningResult);
+        }
+        return winningResult;
+    }
+
+    private Map<WinningType, Integer> createWinningResult() {
+        return Arrays.stream(WinningType.values())
+                .collect(toMap(identity(), type -> 0));
+    }
+
+    private void countDealerWinning(final Name name, final Map<WinningType, Integer> winningResult) {
+        WinningType dealerResult = judgePlayerResult(name).reverse();
+        winningResult.put(dealerResult, winningResult.get(dealerResult) + 1);
+    }
+
+    private WinningType judgePlayerResult(final Name name) {
+        int playerScore = playerScores.get(name);
+        if (playerScore > BLACK_JACK) {
+            return DEFEAT;
+        }
+        if (dealerScore > BLACK_JACK) {
+            return WIN;
+        }
+        if (playerScore > dealerScore) {
+            return WIN;
+        }
+        if (playerScore < dealerScore) {
+            return DEFEAT;
+        }
+        return DRAW;
+    }
```

Comment:
```suggestion
    private WinningType judgePlayerResult(final Name name) {
        if (hasBust(name)) {
            return judgeBustCase(name);
        }
        return judgeNormalCase(name);
    }

    private boolean hasBust(final Name name) {
        return dealerScore > BLACK_JACK || playerScores.get(name) > BLACK_JACK;
    }

    private WinningType judgeBustCase(final Name name) {
        if (playerScores.get(name) > BLACK_JACK) {
            return DEFEAT;
        }
        return WIN;
    }
    
    private WinningType judgeNormalCase(final Name name) {
        if (playerScores.get(name) > dealerScore) {
            return WIN;
        }
        if (playerScores.get(name) < dealerScore) {
            return DEFEAT;
        }
        return DRAW;
    }
```

개인적인 취향인데 이렇게 메서드를 추출해볼 수도 있을 것 같습니다!